### PR TITLE
NEXT-39787 - Google analytics tracking acceptance test

### PR DESCRIPTION
**Pull request by frobel**
### 1. Why is this change necessary?
Extend the acceptance test suite of shopware

### 2. What does this change do, exactly?
Test the google analytics tracking option with an acceptance test

### 5. Checklist

- [ ] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfill them.

### CODE REVIEW

**Comment by frobel** [comment]
@jc-sw-git I separated the tests and added one more regarding declining.

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    await StorefrontHome.consentAcceptAllCookiesButton.click();
+
+    let allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.length).toEqual(2);
+
+    await StorefrontHome.page.reload();
+
+    allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.length).toEqual(4);
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]: To enhance the readability and structure of test cases, adding test.step can be a great approach. It helps to break down the test into logical, understandable steps, making it easier for others to read and maintain. It'd also be great that, people will understand what the test case is doing in detail without understanding the code.

```suggestion
    await test.step('Set system configuration to accept all cookies', async () => {
    await TestDataService.createSystemConfigEntry('core.basicInformation.acceptAllCookies', true, DefaultSalesChannel.salesChannel.id);
});

await test.step('Navigate to the homepage and verify cookie consent banner', async () => {
    await ShopCustomer.goesTo(StorefrontHome.url());
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
});

await test.step('Accept all cookies', async () => {
    await StorefrontHome.consentAcceptAllCookiesButton.click();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(2);  // Verify initial cookies
});

await test.step('Reload page and verify cookies', async () => {
    await StorefrontHome.page.reload();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(4);  // Verify cookies after reload
});

await test.step('Verify cookie consent banner is no longer visible', async () => {
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled').value).toEqual('1');
+    ShopCustomer.expects(allCookies.length).toEqual(5);
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
+
+    await StorefrontHome.page.reload();
+    allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled').value).toEqual('1');
+    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled').value).toEqual('1');
+    ShopCustomer.expects(allCookies.length).toEqual(5);
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts]:
```suggestion
    await test.step('Set up sales channel analytics and navigate to homepage', async () => {
    const salesChannelAnalytics = await TestDataService.createSalesChannelAnalytics();
    await TestDataService.assignSalesChannelAnalytics(DefaultSalesChannel.salesChannel.id, salesChannelAnalytics.id);

    await ShopCustomer.goesTo(StorefrontHome.url());
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
});

await test.step('Configure cookie consent settings', async () => {
    await StorefrontHome.consentConfigureButton.click();
    await StorefrontHome.consentDialogStatisticsCheckbox.click();
    await StorefrontHome.consentDialogMarketingdCheckbox.click();
    await StorefrontHome.consentDialogSaveButton.click();
});

await test.step('Verify cookies are set correctly after consent', async () => {
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled').value).toEqual('1');
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled').value).toEqual('1');
    ShopCustomer.expects(allCookies.length).toEqual(5);
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});

await test.step('Reload page and verify cookies persist', async () => {
    await StorefrontHome.page.reload();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled').value).toEqual('1');
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled').value).toEqual('1');
    ShopCustomer.expects(allCookies.length).toEqual(5);
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled')).not.toBeDefined();
+    ShopCustomer.expects(allCookies.length).toEqual(3);
+
+    await StorefrontHome.page.reload();
+    allCookies = await StorefrontHome.page.context().cookies();
+
+    ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled')).not.toBeDefined();
+    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled')).not.toBeDefined();
+    ShopCustomer.expects(allCookies.length).toEqual(3);
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts]:
```suggestion
    await test.step('Set up sales channel analytics and verify cookie banner', async () => {
    const salesChannelAnalytics = await TestDataService.createSalesChannelAnalytics();
    await TestDataService.assignSalesChannelAnalytics(DefaultSalesChannel.salesChannel.id, salesChannelAnalytics.id);

    await ShopCustomer.goesTo(StorefrontHome.url());
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
});

await test.step('Verify default cookie consent settings', async () => {
    await StorefrontHome.consentConfigureButton.click();
    await ShopCustomer.expects(StorefrontHome.consentDialogTechnicallyRequiredCheckbox).toBeChecked();
    await ShopCustomer.expects(StorefrontHome.consentDialogStatisticsCheckbox).not.toBeChecked();
    await ShopCustomer.expects(StorefrontHome.consentDialogMarketingdCheckbox).not.toBeChecked();
    await ShopCustomer.expects(StorefrontHome.consentDialog.getByRole('checkbox')).toHaveCount(4);
    await StorefrontHome.consentDialogSaveButton.click();
});

await test.step('Verify cookies after saving default consent settings', async () => {
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled')).not.toBeDefined();
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled')).not.toBeDefined();
    ShopCustomer.expects(allCookies.length).toEqual(3);
});

await test.step('Reload page and verify cookie persistence', async () => {
    await StorefrontHome.page.reload();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled')).not.toBeDefined();
    ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled')).not.toBeDefined();
    ShopCustomer.expects(allCookies.length).toEqual(3);
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});
```

File: tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
+    await StorefrontHome.consentConfigureButton.click();
+    await StorefrontHome.offcanvasBackdrop.click();
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
+
+    const allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.length).toEqual(2);
+
+    await productListItemLocators.productImage.click();
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts]:
```suggestion
    await test.step('Set system configuration and create test data', async () => {
    await TestDataService.createSystemConfigEntry('core.basicInformation.acceptAllCookies', true, DefaultSalesChannel.salesChannel.id);
    const product = await TestDataService.createBasicProduct();
    const category = await TestDataService.createCategory();
    await TestDataService.assignProductCategory(product.id, category.id);
});

await test.step('Navigate to homepage and verify cookie banner', async () => {
    await ShopCustomer.goesTo(StorefrontHome.url());
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
});

await test.step('Interact with cookie consent banner', async () => {
    await StorefrontHome.consentConfigureButton.click();
    await StorefrontHome.offcanvasBackdrop.click();
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});

await test.step('Verify cookies after dismissing banner', async () => {
    const allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(2);
});

await test.step('Navigate to product page and verify cookie banner', async () => {
    const productListItemLocators = await StorefrontHome.getListingItemByProductId(product.id);
    await productListItemLocators.productImage.click();
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
});
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAvailability.spec.ts [hunk 5]
```diff
@@ -15,53 +12,21 @@ test('As a shop customer, I want use a basic cookie consent banner in storefront
     await ShopCustomer.expects(StorefrontHome.consentConfigureButton).toBeVisible();
     await ShopCustomer.expects(StorefrontHome.consentCookiePermissionContent).toContainText('This website uses cookies to ensure the best experience possible.');
 
+    let allCookies = await StorefrontHome.page.context().cookies();
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Settings/BasicConsentBannerAvailability.spec.ts]: can you please add test.step for other tests

```suggestion
    await test.step('Navigate to homepage and verify initial cookie banner visibility', async () => {
    await ShopCustomer.goesTo(StorefrontHome.url());
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).not.toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentOnlyTechnicallyRequiredButton).toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentConfigureButton).toBeVisible();
    await ShopCustomer.expects(StorefrontHome.consentCookiePermissionContent).toContainText(
        'This website uses cookies to ensure the best experience possible.'
    );
});

await test.step('Verify initial cookies before any consent', async () => {
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(2);
});

await test.step('Configure cookie settings', async () => {
    await StorefrontHome.consentConfigureButton.click();
    await ShopCustomer.expects(StorefrontHome.consentDialogTechnicallyRequiredCheckbox).toBeChecked();
    await ShopCustomer.expects(StorefrontHome.consentDialog.getByRole('checkbox')).toHaveCount(2);
    await StorefrontHome.consentDialogSaveButton.click();
});

await test.step('Verify cookies after saving consent settings', async () => {
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(3);
});

await test.step('Reload page and verify cookie persistence', async () => {
    await StorefrontHome.page.reload();
    let allCookies = await StorefrontHome.page.context().cookies();
    ShopCustomer.expects(allCookies.length).toEqual(3);
    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
});
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+await test.step('Reload page and verify cookies', async () => {
+    await StorefrontHome.page.reload();
+    let allCookies = await StorefrontHome.page.context().cookies();
+    ShopCustomer.expects(allCookies.length).toEqual(4);  // Verify cookies after reload
+});
+
+await test.step('Verify cookie consent banner is no longer visible', async () => {
+    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
+});
```
Comment (vanpham-sw) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]: please format these lines

**Review by vanpham-sw — APPROVED** [review]

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a shop customer, I want to use a accept all cookies button in the basic cookie consent banner in storefront.', { tag: '@Settings' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]:
```suggestion
test('As a shop customer, I want to use an "Accept All Cookies" button in the basic cookie consent banner in the storefront.', { tag: '@Settings' }, async ({
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    ShopCustomer,
+    StorefrontHome,
+    TestDataService,
+    DefaultSalesChannel,
+}) => {
+
+    await test.step('Set system configuration to enable accept all cookies button', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]:
```suggestion
    await test.step('Enable "Accept All Cookies" button in system configuration', async () => {
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        await TestDataService.createSystemConfigEntry('core.basicInformation.acceptAllCookies', true, DefaultSalesChannel.salesChannel.id);
+    });
+
+    await test.step('Navigate to the homepage and verify cookie consent banner', async () => {
+        await ShopCustomer.goesTo(StorefrontHome.url());
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
+        await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
+    });
+
+    await test.step('Accept all cookies', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]:
```suggestion
    await test.step('Click "Accept All Cookies" and verify initial cookies', async () => {
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        await StorefrontHome.consentAcceptAllCookiesButton.click();
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.length).toEqual(2);  // Verify initial cookies
+    });
+
+    await test.step('Reload page and verify cookies', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]:
```suggestion
    await test.step('Reload page and verify additional cookies are set', async () => {
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    await test.step('Navigate to the homepage and verify cookie consent banner', async () => {
+        await ShopCustomer.goesTo(StorefrontHome.url());
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
+        await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
+    });
+
+    await test.step('Accept all cookies', async () => {
+        await StorefrontHome.consentAcceptAllCookiesButton.click();
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.length).toEqual(2);  // Verify initial cookies
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]: You can remove the comment here, If you apply my test step above.

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    });
+
+    await test.step('Reload page and verify cookies', async () => {
+        await StorefrontHome.page.reload();
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.length).toEqual(4);  // Verify cookies after reload
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]: You can remove the comment here, If you apply my test step above.

File: tests/acceptance/tests/Settings/BasicConsentBannerAvailability.spec.ts [hunk 13]
```diff
@@ -1,67 +1,42 @@
 import { test } from '@fixtures/AcceptanceTest';
 
-// Annotate entire file as serial run.
-test.describe.configure({ mode: 'serial' });
-
 test('As a shop customer, I want use a basic cookie consent banner in storefront.', { tag: '@Settings' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAvailability.spec.ts]:
```suggestion
test('As a shop customer, I want use a basic cookie consent banner in the storefront.', { tag: '@Settings' }, async ({
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAvailability.spec.ts [hunk 14]
```diff
@@ -1,67 +1,42 @@
...
-    await ShopCustomer.expects(StorefrontHome.consentCookiePermissionContent).toContainText('This website uses cookies to ensure the best experience possible.');
-
-    await StorefrontHome.consentConfigureButton.click();
-    await ShopCustomer.expects(StorefrontHome.consentDialogTechnicallyRequiredCheckbox).toBeChecked();
-    await StorefrontHome.consentDialogSaveButton.click();
-
-    await StorefrontHome.page.reload();
-    await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
-
+    await test.step('Navigate to homepage and verify initial cookie banner visibility', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerAvailability.spec.ts]:
```suggestion
    await test.step('Navigate to homepage and verify initial cookie banner visibility and content', async () => {
```

File: tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts [hunk 15]
```diff
@@ -0,0 +1,37 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a shop customer, I want to continue shopping without accepting the cookies in storefront.', { tag: '@Settings' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts]:
```suggestion
test('As a shop customer, I want to continue shopping without accepting the cookies in the storefront.', { tag: '@Settings' }, async ({
```

File: tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+    const category = await TestDataService.createCategory();
+    await TestDataService.assignProductCategory(product.id, category.id);
+
+    await test.step('Navigate to homepage and verify cookie banner', async () => {
+        await ShopCustomer.goesTo(StorefrontHome.url());
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
+        await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
+    });
+
+    await test.step('Interact with cookie consent banner', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts]:
```suggestion
    await test.step('Dismiss cookie banner using the configure option', async () => {
```

File: tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+        await StorefrontHome.consentConfigureButton.click();
+        await StorefrontHome.offcanvasBackdrop.click();
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
+    });
+
+    await test.step('Verify cookies after dismissing banner', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts]:
```suggestion
    await test.step('Verify cookies after dismissing the cookie banner', async () => {
```

File: tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.length).toEqual(2);
+    });
+
+    await test.step('Navigate to product page and verify cookie banner', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/BasicConsentBannerDeclineCookies.spec.ts]:
```suggestion
    await test.step('Navigate to the product page and verify the cookie banner', async () => {
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts [hunk 19]
```diff
@@ -0,0 +1,41 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a shop customer, I want to accept the Google Analytics tracking within basic cookie consent banner in storefront.', { tag: '@Settings' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts]:
```suggestion
test('As a shop customer, I want to accept Google Analytics tracking via the basic cookie consent banner in the storefront.', { tag: '@Settings' }, async ({
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+    ShopCustomer,
+    StorefrontHome,
+    TestDataService,
+    DefaultSalesChannel,
+}) => {
+
+    await test.step('Set up sales channel analytics and navigate to homepage', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts]:
```suggestion
    await test.step('Configure sales channel analytics and verify cookie banner visibility on the home page', async () => {
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+        const salesChannelAnalytics = await TestDataService.createSalesChannelAnalytics();
+        await TestDataService.assignSalesChannelAnalytics(DefaultSalesChannel.salesChannel.id, salesChannelAnalytics.id);
+
+        await ShopCustomer.goesTo(StorefrontHome.url());
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
+    });
+
+    await test.step('Configure cookie consent settings', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts]:
```suggestion
    await test.step('Enable Google Analytics and marketing tracking via cookie settings', async () => {
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+        await StorefrontHome.consentConfigureButton.click();
+        await StorefrontHome.consentDialogStatisticsCheckbox.click();
+        await StorefrontHome.consentDialogMarketingdCheckbox.click();
+        await StorefrontHome.consentDialogSaveButton.click();
+    });
+
+    await test.step('Verify cookies are set correctly after consent', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts]:
```suggestion
    await test.step('Verify tracking cookies are set correctly after consent', async () => {
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled').value).toEqual('1');
+        ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled').value).toEqual('1');
+        ShopCustomer.expects(allCookies.length).toEqual(5);
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).not.toBeVisible();
+    });
+
+    await test.step('Reload page and verify cookies persist', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsAcceptTracking.spec.ts]:
```suggestion
    await test.step('Verify tracking cookies persist after page reload', async () => {
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts [hunk 24]
```diff
@@ -0,0 +1,42 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a shop customer, I want to accept the technical required cookies without activating Google Analytics tracking within basic cookie consent banner in storefront.', { tag: '@Settings' }, async ({
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts]:
```suggestion
test('As a shop customer, I want to accept only the technically required cookies without activating Google Analytics tracking via the basic cookie consent banner in the storefront.', { tag: '@Settings' }, async ({
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+    ShopCustomer,
+    StorefrontHome,
+    TestDataService,
+    DefaultSalesChannel,
+}) => {
+
+    await test.step('Set up sales channel analytics and verify cookie banner', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts]:
```suggestion
    await test.step('Configure sales channel analytics and verify cookie banner visibility', async () => {
```

File: tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    await test.step('Verify cookies after saving default consent settings', async () => {
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.find(c => c.name == 'google-analytics-enabled')).not.toBeDefined();
+        ShopCustomer.expects(allCookies.find(c => c.name == 'google-ads-enabled')).not.toBeDefined();
+        ShopCustomer.expects(allCookies.length).toEqual(3);
+    });
+
+    await test.step('Reload page and verify cookie persistence', async () => {
```
Comment (yusufttur) [inline tests/acceptance/tests/Settings/GoogleAnalyticsDeclineTracking.spec.ts]:
```suggestion
    await test.step('Verify cookies persist after page reload', async () => {
```

**Review by yusufttur — CHANGES_REQUESTED** [review]
I suggest some changes to the descriptions and the steps to better readability.

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+    await test.step('Navigate to the homepage and verify cookie consent banner', async () => {
+        await ShopCustomer.goesTo(StorefrontHome.url());
+        await ShopCustomer.expects(StorefrontHome.consentCookieBannerContainer).toBeVisible();
+        await ShopCustomer.expects(StorefrontHome.consentAcceptAllCookiesButton).toBeVisible();
+    });
+
+    await test.step('Click "Accept All Cookies" and verify initial cookies', async () => {
+        await StorefrontHome.consentAcceptAllCookiesButton.click();
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.length).toEqual(2);  // Verify initial cookies
```
Comment (frobel) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]:
```suggestion
        ShopCustomer.expects(allCookies.length).toEqual(2);
```

File: tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts [hunk 28]
```diff
@@ -0,0 +1,35 @@
...
+    });
+
+    await test.step('Reload page and verify additional cookies are set', async () => {
+        await StorefrontHome.page.reload();
+        const allCookies = await StorefrontHome.page.context().cookies();
+        ShopCustomer.expects(allCookies.length).toEqual(4);  // Verify cookies after reload
```
Comment (frobel) [inline tests/acceptance/tests/Settings/BasicConsentBannerAcceptAllCookies.spec.ts]:
```suggestion
        ShopCustomer.expects(allCookies.length).toEqual(4);
```

**Review by yusufttur — APPROVED** [review]

**Review by GitEvil — APPROVED** [review]

**Pull request merged**